nemo/collections/nlp/models/language_modeling/megatron_glue_model.py

```diff
@@ -0,0 +1,269 @@
+            self.cfg.data.train_ds.file_path,
+            task_name=self.cfg.data.train_ds.task_name,
+            tokenizer=self.model.tokenizer,
+            max_seq_length=self.cfg.data.train_ds.max_seq_length,
+        )
+        self._validation_ds = TextToTextGLUEDataset(
+            self.cfg.data.validation_ds.file_path,
+            task_name=self.cfg.data.validation_ds.task_name,
+            tokenizer=self.model.tokenizer,
+            max_seq_length=self.cfg.data.validation_ds.max_seq_length,
+        )
+        logging.info(f'Length of train dataset: {len(self._train_ds)}')
+        logging.info(f'Length of val dataset: {len(self._validation_ds)}')
+        logging.info(f'Length of test dataset: {len(self._test_ds)}')
+        logging.info(f'Finished building T5 datasets.')
+        return self._train_ds, self._validation_ds, self._test_ds
+
+    def build_pretraining_data_loader(self, dataset, batch_size, shuffle, num_workers, pin_memory):
+        """Buld dataloader given an input dataset."""
+
+        if dataset is None:
+            return None
+
+        rank = parallel_state.get_data_parallel_rank()
+        world_size = parallel_state.get_data_parallel_world_size()
+        sampler = torch.utils.data.distributed.DistributedSampler(
+            dataset, num_replicas=world_size, rank=rank, shuffle=shuffle
+        )
+
+        # Data loader. Note that batch size is the per GPU batch size.
+        return torch.utils.data.DataLoader(
+            dataset,
+            collate_fn=dataset.collate_fn,
+            batch_size=batch_size,
+            sampler=sampler,
+            num_workers=num_workers,
+            pin_memory=pin_memory,
+            drop_last=False,
+        )
+
+    def setup(self, stage=None):
+        if stage == 'predict':
+            return
+        self.build_train_valid_test_datasets(test_only=stage == 'test')
+        self.setup_test_data()
+        if stage == 'test':
+            return
+        self.setup_training_data()
+        self.setup_validation_data()
+
+    def setup_training_data(self):
+        self._train_dl = self.build_pretraining_data_loader(
+            self._train_ds,
+            self.cfg.data.train_ds.batch_size,
+            shuffle=True,
+            num_workers=self.cfg.data.train_ds.num_workers,
+            pin_memory=True,
+        )
+
+    def setup_validation_data(self):
+        self._validation_dl = self.build_pretraining_data_loader(
+            self._validation_ds,
+            self.cfg.data.validation_ds.batch_size,
+            shuffle=False,
+            num_workers=self.cfg.data.validation_ds.num_workers,
+            pin_memory=True,
+        )
+
+    def setup_test_data(self):
+        self._test_dl = self.build_pretraining_data_loader(
+            self._test_ds,
+            self.cfg.data.test_ds.batch_size,
+            shuffle=False,
+            num_workers=self.cfg.data.test_ds.num_workers,
+            pin_memory=True,
+        )
+
```

Comment:
In your `megatron_t5_model.py` file, the `list_available_models` is class method, make it consistent?

Reply:
Fixed!